### PR TITLE
Disable nrvo mir opt

### PR DESCRIPTION
**Pull request by JakobDegen**
See #111005 and #110902 . The ICE can definitely be hit on stable, the miscompilation I'm not sure about. The pass makes some pretty sketchy assumptions though, and we should not have it on while that's the case.

I'm not going to work on actually fixing this, it's probably not excessively difficult though.

r? rust-lang/mir-opt

### CODE REVIEW

**Comment by cjgillot** [comment]
Should we remove this pass entirely, and let dest-prop do it instead?

**Comment by JakobDegen** [comment]
If it were on by default, yes. Until then, I'm not so sure

File: tests/mir-opt/nrvo_miscompile_111005.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+pub fn wrong(arg: char) -> char {
+    mir!({
+        let temp = arg;
+        RET = temp;
+        temp = 'b';
+        Return()
+    })
+}
+
+fn main() {}
```
Comment (cjgillot) [inline tests/mir-opt/nrvo_miscompile_111005.rs:19]: Could you add the EMIT_MIR directive?

Comment (JakobDegen) [reply tests/mir-opt/nrvo_miscompile_111005.rs]: /me wonders how I managed to forget that

**Comment by nnethercote** [comment]
Lots of regressions here. It sure would be nice if these could be regained, via dest-prop or any other means.

**Pull request merged**